### ISSUE TITLE
How do I ignore satellite assembly when publishing?

### ISSUE THREAD

**Issue by Nirmal4G** [NuGet/Home#8359]
From: NuGet/docs.microsoft.com-nuget#1549

I want to reduce the my size of my published project. For local testing, I don't need all the language specific resources, just the neutral version is more than enough. I also test them manually by removing and adding language specific resources when the project is running!

So, How do I ignore the language folders from NuGet `PackageReference`?

**edit by @nkolev92**

Related to https://github.com/NuGet/Home/issues/5986

**Comment by Nirmal4G** [comment]
There are no options for excluding resources in `*Assets` metadata on `Framework/Project/PackageReference`.

In addition to existing options (like compile, runtime, etc...), I would like a `resource` option, that can handle resources.

**Comment by heng-liu** [comment]
Hi, thanks for reporting this issue!
Do you mean for now, you put localized resource assemblies in a single package? If yes, have you tried to use [localized satellite packages ](https://docs.microsoft.com/en-us/nuget/create-packages/creating-localized-packages#localized-satellite-packages)?

**Comment by Nirmal4G** [comment]
No, it's how I consume a single/meta package with satellite dlls.

For e.g. If you put `ExcludeAssets="Runtime"` the runtime assets (`native` folder) from the referenced package, will be excluded from the projects' restore deps graph. But there is no option for resources!

**Comment by heng-liu** [comment]
Will using localized satellite packages solve your problem? It will separate localized resources and IntelliSense XML files into satellite packages. You may check the two different approaches here: https://docs.microsoft.com/en-us/nuget/create-packages/creating-localized-packages

**Comment by Nirmal4G** [comment]
I think you misunderstood my problem.

**I'm not the one creating those packages, @Microsoft is**. So, If they separate resources in to different packages, then, yes, that would solve my problem.

Roslyn team have packages with `Microsoft.CodeAnalysis*` ID. Those package have satellite assemblies within a single package, not following the guideline you mentioned above and as do so many other packages.

My issue comes down to this, Would you guys change your packaging style just for me and few others? **NO**.

We can currently include and exclude specific asset types in `PackageReference` through `IncludeAssets`, `PrivateAssets`, `ExcludeAssets`. They have `compile`, `runtime`, `build`, `content`, etc... but no `resource` type for satellite assemblies.

I have no need for other languages in my test deployment. So I want to exclude all the resources, that I produce and the resources that I get from the packages. **How do I do that?**

**Comment by ImrePyhvel** [comment]
The issue is still relevant. 

I do want my own satellite assemblies to be included for localization, but I need to exclude SOME satellite assemblies from dependencies, for example from https://www.nuget.org/packages/Microsoft.Data.SqlClient which packs in additional 10x 250KB files. My app has nothing to do with most languages forced on me (chinese, korean, whatnot.. ), nor do I want technical error messages in our logs to be in languages the support team does not understand. I do not want localization for this specific dependency.

Yes, it should be thought of by package authors, but it would be naive to hope every author would and packages in the wild prove they don't. Hence package consumers should have the option to choose which resources to include on build with greater control and exclude all sattellite assemblies when needed.

It would be sufficient if we were able to exclude explicit assets by path. This way nuget would not have to know about the concept of "satellite assemblies".
